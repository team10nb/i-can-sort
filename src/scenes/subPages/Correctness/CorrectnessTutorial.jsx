/*
    Author: Yuting Jiang, Yijie Lu
*/

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MobileStepper from '@material-ui/core/MobileStepper';
import Button from '@material-ui/core/Button';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Chip from '@material-ui/core/Chip';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import HomeIcon from '@material-ui/icons/Home';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
import {color} from '../../mainPages/Correctness';
import Input from '../../../components/InputTutorial/InputTutorial';
import Example from './Tutorial/Example';
import Partial_Def from './Tutorial/Parital_Def';
import Total_Def from './Tutorial/Total_Def';


const drawerWidth = 240;
const useStyles = makeStyles((theme) => ({

    appBar1: {
        [theme.breakpoints.up('sm')]: {
            width: `calc(100% - ${drawerWidth}px)`,
            marginLeft: drawerWidth,
          },
        height:45,
        backgroundColor:color,       
    },

    appBar2: {      
        top: 'auto',
        bottom: 0,
        [theme.breakpoints.up('sm')]: {
            width: `calc(100% - ${drawerWidth}px)`,
            marginLeft: drawerWidth,
          },
    },

    div:{
        display:"flex",
        alignItems:"center",       
        '& > *': {
          margin: theme.spacing(0),
        },
    },

    icon:{
        marginRight: theme.spacing(2),
        width: 100,
        height: 45,
    },

    chip:{
        position: 'relative',
        marginRight: -50,
        marginLeft: 120,
        minWidth: 400,
        justifyContent:"center",  
        fontWeight: "600",      
        color: color,
        fontFamily:"Quicksand",
        backgroundColor: "#F8F8F8",
    },

    stepper:{        
        alignItems:"bottom",
        fontSize: 16,
        fontWeight: "bold",        
        color: color,
        backgroundColor: "#F0F0F0",     
    },

    content:{
        display:"flex",
        alignItems:"center",          
        width: 1020,
        height: 560,
    },

    drawer: {
        width: drawerWidth,   
        flexShrink: 0,
        height:620,
        fontFamily: "Arial"
    },
    
    drawerPaper: {
        width: drawerWidth,    
    },

    drawerContainer: {
        overflow: 'auto',   
    },

    listHeader:{
        height:30,       
        color: color,
        fontSize: "20px"
    },

    listItem:{
        height:40,
    },

    hightlightItem:{
        height:40,
        backgroundColor: color,
        color: "#F8F8F8",    
        "&:hover": {backgroundColor:color},
    },
    listItemText:{
        fontFamily:"Quicksand",
        fontWeight: "600",
      },
      listItemTitle:{
        fontFamily:"Quicksand",
        fontWeight: "600",
        fontSize: "18px"
      },
      listItemCatalogue:{
        fontFamily:"Quicksand",
        fontWeight: "600",
        fontSize: "20px"
      }

    
   
  }));

  function getSteps() {
    return [
            'Termination',
            'Introduction',
            'Partial Correctness',
            'Total Correctness',
            'Example'
    ];
  }

  function getStepContent(activeStep) {
      switch (activeStep) {
          case 1:
            return(
                <Input />
            );
        case 2:
            return(
                <Partial_Def />
            );
        case 3:
            return(
                <Total_Def />
            );
        case 4:
            return(
                <Example />
            );
        default:
            break;
      }
  }

  
    export default function CorrectnessTutorial(props) {
    const classes = useStyles();

    const [activeStep, setActiveStep] = React.useState(0);

    const maxSteps = getSteps().length;

    const step = getSteps();

    const handleNext = () => {
        setActiveStep((prevActiveStep) => prevActiveStep + 1);
    };

    const handleBack = () => {
        setActiveStep((prevActiveStep) => prevActiveStep - 1);
    };

    const handleChange =(index) => {
        setActiveStep(index);
    };
   
    const handleClick = () => {
        //return to procedure main page with the progress
        props.history.push({pathname: '/CorrectnessMainPage'});
    };

    

    return (
        <div>
            <AppBar className={classes.appBar1} >
                <div className ={classes.div}>
                <Button className={classes.icon} onClick={handleClick}>
                    <HomeIcon style={{ fontSize: 30, color: "#FFFFFF"}} />
                </Button>
                <Chip className={classes.chip} label= {step[activeStep]} />  
                </div>                          
            </AppBar>
     
            <main className={classes.content}>
                <Drawer
                className={classes.drawer}
                variant="permanent"
                classes={{
                paper: classes.drawerPaper,
                }}
                >
                    <div className={classes.drawerContainer}>
                        <List>
                            <ListItem  className = {classes.listHeader}>                
                            <ListItemText primary= 'Catalogue' classes={{primary:classes.listItemCatalogue}}/>
                            </ListItem>
                        </List>
{/*                         <Divider /> */}
{/*                         <List > */}
{/*                             <ListItem  className = {classes.listItem}>                           */}
{/*                             <ListItemText primary= 'Input' /> */}
{/*                             </ListItem> */}
{/*                             {['Input'].map((text, index) => ( */}
{/*                                  */}
{/*                             index === activeStep  */}
{/*                                 ?<ListItem className = {classes.hightlightItem} button key={text} onClick={() => handleChange(index)}> */}
{/*                                     <ListItemIcon><ChevronRightIcon/></ListItemIcon> */}
{/*                                     <ListItemText primary={text} /> */}
{/*                                 </ListItem> */}
{/*                                 :<ListItem className = {classes. listItem} button key={text} onClick={() => handleChange(index)}> */}
{/*                                     <ListItemIcon><ChevronRightIcon/></ListItemIcon> */}
{/*                                     <ListItemText primary={text} /> */}
{/*                                 </ListItem>  */}
{/*                             ))}                   */}
{/*                         </List> */}
{/*                         <Divider /> */}
                        <List>
                            <ListItem className = {classes.listItem}>             
                            <ListItemText primary= 'Termination' classes={{primary:classes.listItemTitle}}/>
                            </ListItem>
                            {['Termination'].map((text, index) => (
                            index === activeStep
                                ?<ListItem  className = {classes.hightlightItem} button key={text} onClick={() => handleChange(index )}>
                                    <ListItemIcon><ChevronRightIcon/></ListItemIcon>
                                    <ListItemText primary={text} classes={{primary:classes.listItemText}}/>
                                </ListItem>
                                :<ListItem className = {classes. listItem} button key={text} onClick={() => handleChange(index )}>
                                    <ListItemIcon><ChevronRightIcon/></ListItemIcon>
                                    <ListItemText primary={text} classes={{primary:classes.listItemText}}/>
                                </ListItem> 
                            ))}
                        </List>

                        <Divider />
                        <List>
                            <ListItem className = {classes.listItem}>                           
                            <ListItemText primary= 'Correctness' classes={{primary:classes.listItemTitle}}/>
                            </ListItem>
                            {['Introduction' , 'Partial Correctness', 'Total Correctness', 'Example' ].map((text, index) => (
                            index + 1 === activeStep
                            ?<ListItem  className = {classes.hightlightItem} button key={text} onClick={() => handleChange(index + 1)}>
                                <ListItemIcon><ChevronRightIcon/></ListItemIcon>
                                <ListItemText primary={text} classes={{primary:classes.listItemText}}/>
                            </ListItem>
                            :<ListItem className = {classes. listItem} button key={text} onClick={() => handleChange(index + 1)}
                                ><ListItemIcon><ChevronRightIcon/></ListItemIcon>
                                <ListItemText primary={text} classes={{primary:classes.listItemText}}/>
                            </ListItem> 
                            ))}
                        </List>
                   </div>
                </Drawer>
                {getStepContent(activeStep)}
            </main>



            <AppBar  className = {classes.appBar2}>
                <MobileStepper
                    steps={maxSteps}
                    position="static"
                    className = {classes.stepper}
                    variant="text"                  
                    activeStep={activeStep}
                    nextButton={
                    <Button size="small" variant="contained" style={{ backgroundColor: color, color: "#FFFFFF"}} onClick={handleNext} disabled={activeStep === maxSteps - 1}>
                        Next                       
                    </Button>
                    }
                    middlebutton = {<MoreHorizIcon color={color} />}
                    backButton={
                    <Button size="small" variant="contained" style={{ backgroundColor: color, color: "#FFFFFF"}} onClick={handleBack} disabled={activeStep === 0}>
                        Back
                    </Button>
                    }
                />
            </AppBar>
        </div>
       
       
    );
}

