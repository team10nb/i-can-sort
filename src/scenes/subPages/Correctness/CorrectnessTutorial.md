
### **Developer**
Yijie Lu

Reference: MATERIAL-UI (https://material-ui.com)



###  **Introduction**

This is a correctness tutorial. Users can get information about correctness from here.

###  **Import**

```html
import { makeStyles } from '@material-ui/core/styles';
```

###  **Components**

| Name | Description | Child components |
| ---- | ----------- | ---------------- |
| AppBar1 |    a bar contains each subpage's name     |       \          |   
| Drawer |    a drawer show the level of subpage     |       List        |
| List   |    a list of each subpage's name         |       \    |
| AppBar1 |   a bar show page number and give access to the last and the next page|  \ |   

###  **SubPages**

| Name   | Description | 
| ----   | ----------- | 
| Termination         | two examples show the termination knowledge |     
| Introduction        |     introduction of correctness           |
| Partial Correctness |   knowledge of partial correctness        | 
| Total Correctness   |    knowledge of total correctness         |
| Example             | three examples of each algorithm type     |
