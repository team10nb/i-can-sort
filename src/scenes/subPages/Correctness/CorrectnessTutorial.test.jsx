
import { render } from '@testing-library/react';
import CorrectnessTutorial from './CorrectnessTutorial';

jest.mock("../../../components/InputTutorial/InputTutorial", () => {
    return {
        __esModule: true,
        default: () => <div>mockInputTutorial</div>,
    }
});

test('should contain buttons', () => {
    const {getAllByText} = render(        
            <CorrectnessTutorial />
    );
    // getAllByText("Input");
    // getAllByText("Input_Def");
    getAllByText("Termination");
    // getAllByText("Termination_Def");
    getAllByText("Introduction");
    getAllByText("Partial Correctness");
    getAllByText("Total Correctness");
    // getAllByText("Partial_Example");
    getAllByText("Example");
    
    // getAllByText("Correctness_Intro");

    getAllByText("Next");
    getAllByText("Back");
})



// test('should match Snapshot', () => {
//     const content = render(<CorrectnessTutorial />);
//     expect(content).toMatchSnapshot();
// })

