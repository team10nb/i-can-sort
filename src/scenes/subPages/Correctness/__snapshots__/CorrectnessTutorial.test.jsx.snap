// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar1-14 mui-fixed MuiPaper-elevation4"
        >
          <div
            class="makeStyles-div-16"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-icon-17"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="font-size: 30px; color: rgb(255, 255, 255);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiChip-root makeStyles-chip-18"
            >
              <span
                class="MuiChip-label"
              >
                Input
              </span>
            </div>
          </div>
        </header>
        <main
          class="makeStyles-content-20"
        >
          <div
            class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-21"
          >
            <div
              class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-22 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
            >
              <div
                class="makeStyles-drawerContainer-23"
              >
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <li
                    class="MuiListItem-root makeStyles-listHeader-24 MuiListItem-gutters"
                  >
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Catalogue
                      </span>
                    </div>
                  </li>
                </ul>
                <hr
                  class="MuiDivider-root"
                />
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <li
                    class="MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters"
                  >
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Input
                      </span>
                    </div>
                  </li>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root makeStyles-hightlightItem-26 MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Input
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </ul>
                <hr
                  class="MuiDivider-root"
                />
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <li
                    class="MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters"
                  >
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Termination
                      </span>
                    </div>
                  </li>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Termination
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </ul>
                <hr
                  class="MuiDivider-root"
                />
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <li
                    class="MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters"
                  >
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Correctness
                      </span>
                    </div>
                  </li>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Comparison
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Partial Correctness
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Total Correctness
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Conclusion
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </ul>
              </div>
            </div>
          </div>
          <div>
            mockInputTutorial
          </div>
        </main>
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar2-15 mui-fixed MuiPaper-elevation4"
        >
          <div
            class="MuiPaper-root MuiMobileStepper-root MuiMobileStepper-positionStatic makeStyles-stepper-19 MuiPaper-elevation0"
            middlebutton="[object Object]"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
              disabled=""
              style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Back
              </span>
            </button>
            1
             / 
            6
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
              style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Next
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar1-14 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="makeStyles-div-16"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-icon-17"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="font-size: 30px; color: rgb(255, 255, 255);"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiChip-root makeStyles-chip-18"
          >
            <span
              class="MuiChip-label"
            >
              Input
            </span>
          </div>
        </div>
      </header>
      <main
        class="makeStyles-content-20"
      >
        <div
          class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-21"
        >
          <div
            class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-22 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
          >
            <div
              class="makeStyles-drawerContainer-23"
            >
              <ul
                class="MuiList-root MuiList-padding"
              >
                <li
                  class="MuiListItem-root makeStyles-listHeader-24 MuiListItem-gutters"
                >
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Catalogue
                    </span>
                  </div>
                </li>
              </ul>
              <hr
                class="MuiDivider-root"
              />
              <ul
                class="MuiList-root MuiList-padding"
              >
                <li
                  class="MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters"
                >
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Input
                    </span>
                  </div>
                </li>
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-hightlightItem-26 MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Input
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </ul>
              <hr
                class="MuiDivider-root"
              />
              <ul
                class="MuiList-root MuiList-padding"
              >
                <li
                  class="MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters"
                >
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Termination
                    </span>
                  </div>
                </li>
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Termination
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </ul>
              <hr
                class="MuiDivider-root"
              />
              <ul
                class="MuiList-root MuiList-padding"
              >
                <li
                  class="MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters"
                >
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Correctness
                    </span>
                  </div>
                </li>
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Comparison
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Partial Correctness
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Total Correctness
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-25 MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Conclusion
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </ul>
            </div>
          </div>
        </div>
        <div>
          mockInputTutorial
        </div>
      </main>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar2-15 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiPaper-root MuiMobileStepper-root MuiMobileStepper-positionStatic makeStyles-stepper-19 MuiPaper-elevation0"
          middlebutton="[object Object]"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Back
            </span>
          </button>
          1
           / 
          6
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
            style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Next
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
